server/message_hooks_test.go

```diff
@@ -2880,3 +2881,136 @@ func TestUserWillLogin(t *testing.T) {
 		})
 	}
 }
+
+func TestMessageHasBeenPosted(t *testing.T) {
+	assert := require.New(t)
+	th := setupTestHelper(t)
+
+	// TODO: remove this when the issue is fixed in the API
+	getDirectChannel := func(t *testing.T, senderID, receiverID string) *model.Channel {
+		t.Helper()
+
+		channel, appErr := th.p.API.GetDirectChannel(senderID, receiverID)
+		if appErr != nil {
+			if appErr.Id != "app.sharedchannel.dm_channel_creation.internal_error" {
+				assert.Nil(appErr)
+			}
+			channel, _ = th.p.API.GetDirectChannel(senderID, receiverID)
+		}
+		return channel
+	}
+
+	setup := func(t *testing.T, selectiveSync bool) (msTeamsUser, MMUser, syntheticUser *model.User) {
+		th.ResetPreserveDB(t)
```

Comment:
Oops it was just a development need, I should remove this... and/or make it better eventually.

context: I had a failing test, and at the end of the test my container was removed which was unfortunate to go see what hapenned. So I commented the lines that would remove the container + made a reset method that would preserve my DB so I can inspect it.

Ideally, it would be nice to have a way (maybe an env variable?) to instruct the test engine to not remove the database container and not do the `DELETE`s when you want to inspect what's happening but don't want to have to run tons of "debug" queries in the test 😄